src/main/python/tensorframes/core.py

```diff
@@ -37,6 +37,19 @@ def _get_shape(node):
     l = node.get_shape().as_list()
     return [-1 if x is None else x for x in l]
 
+def _initialize_variable(graph, fetches, initial_variable):
+    with tf.Session(graph=graph) as sess:
```

Comment:
Can you add a check for the case that there are no variables to initialize? In that case, you can simply return the graph. I am concerned about mixed clusters for which the driver is a CPU machine, and for which the device placement may fail when loading this graph into the session.